proto/site/v1alpha/site.proto

```diff
@@ -0,0 +1,243 @@
+message RedeemInviteTokenResponse {}
+
+// Request to get the site info.
+message GetSiteInfoRequest {}
+
+// Request to update site info. Doesn't support partial updates.
+message UpdateSiteInfoRequest {
+  // Title of the site.
+  string title = 1;
+
+  // Description of the site.
+  string description = 2;
+}
+
+// Request to list site members.
+message ListMembersRequest {
+  // Optional. Number of items per page.
+  int32 page_size = 1;
+
+  // Optional. Token for a specific page.
+  string page_token = 2;
+}
+
+// Response listing site members.
+message ListMembersResponse {
+  // List of members.
+  repeated Member members = 1;
+
+  // Optional token for the next page.
+  string next_page_token = 2;
+}
+
+// Request to get information about a specific member.
+message GetMemberRequest {
+  // Required. Mintter Account ID to get the information for.
+  string account_id = 1;
+}
+
+// Request to delete an existing member.
+message DeleteMemberRequest {
+  // Required. Mintter Account ID of the member to be deleted.
+  string account_id = 1;
+}
+
+// Request to list blocked accounts.
+message ListBlockedAccountsRequest {
+  // Optional. Number of items per page.
+  int32 page_size = 1;
+
+  // Options. Token for a specific page if any.
+  string next_page_token = 2;
+}
+
+// Response with a list of blocked accounts.
+message ListBlockedAccountsResponse {
+  // List of blocked accounts.
+  repeated string accounts = 1;
+
+  // Token for the next page if any.
+  string next_page_token = 2;
+}
+
+// Invite token is produced by the owner of the site to invite new members.
+message InviteToken {
+  // The value of the invite token.
+  string token = 1;
+
+  // Timestamp after which the token will not be recognized by the site.
+  google.protobuf.Timestamp expire_time = 2;
+}
+
+// Public-facing information about the Mintter Site.
+message SiteInfo {
+  // Hostname under which this site is exposed.
+  string hostname = 1;
+
+  // Title of this site.
+  string title = 2;
+
+  // Description of this site.
+  string description = 3;
+
+  // Mintter Account ID that is configured as the owner of this site.
+  string owner = 4;
+}
+
+// Member of the site.
+message Member {
+  // Roles a member of the Mintter Site can have.
+  enum Role {
+    // Invalid default value.
+    ROLE_UNSPECIFIED = 0;
+
+    // Currently there can only be one owner of the site.
+    OWNER = 1;
+
+    // Editors are allowed to push content to the site.
+    EDITOR = 2;
+  }
+
+  // Mintter Account ID of the member.
+  string account_id = 1;
+
+  // Member's role on the site.
+  Role role = 2;
+}
+
+// Response for the /.well-known discovery HTTP page.
+// TODO: define the actual /.well-known path.
+message DiscoveryConfig {
+  // TODO: add the necessary fields to export under the /.well-known HTTP discovery path.
+}
+
+// Request the server to pin+publish a document.
+message PublishRequest {
+  // id of the doc to publish
+  string doc_id = 1;
```

Comment:
Also, we should include not only the published document but also all the ID's of the referenced documents 
![image](https://user-images.githubusercontent.com/11684004/214337586-0a81c6f3-3dce-464f-8880-1f749f9a929f.png)

Reply:
hmm so we would also send `referenced_doc_ids` or something?

yesterday I thought we agreed upon one-level of references being automatically published, we didn't decide if the client or p2p server is responsible for doing that?

Reply:
@ericvicenti we talked about it last week, I think you weren't there. And we settled on that the easiest would be if the client is responsible to send whatever is needed to properly display the document being published.

Reply:
ok we discussed this, we can provide the referenced ids from the client at publish time

and we will also re-publish to all publications when the user hits save